platform/storage/posix/src/mender-storage.c

```diff
@@ -37,11 +37,20 @@
 #define MENDER_STORAGE_NVS_UPDATE_STATE    CONFIG_MENDER_STORAGE_PATH "um_state.dat"
 #define MENDER_STORAGE_NVS_DEVICE_CONFIG   CONFIG_MENDER_STORAGE_PATH "config.json"
 #define MENDER_STORAGE_NVS_PROVIDES        CONFIG_MENDER_STORAGE_PATH "provides.txt"
+#define MENDER_STORAGE_NVS_ARTICACT_NAME   CONFIG_MENDER_STORAGE_PATH "artifact_name.txt"
 
 mender_err_t
 mender_storage_init(void) {
 
-    /* Nothing to do */
+    /* Get artifact name - set to unknown if not found */
+    char *artifact_name = NULL;
+    if (MENDER_NOT_FOUND == mender_storage_get_artifact_name(&artifact_name)) {
```

Comment:
Instead of writing `"unknown"` to the flash if `MENDER_NOT_FOUND`, can't we just make this function return it? Hence, we would spare the flash for one write?

Reply:
I guess I could make `mender_storage_get_artifact_name` just set artifact_name to unknown on `MENDER_NOT_FOUND`

Reply:
What I mean is that we should avoid unnecessary writes to the flash, so that we don't ware it out